microsoft/testsuites/dpdk/dpdksuite.py

```diff
@@ -81,28 +68,79 @@ def _execute_expect_zero(self, node: Node, cmd: str) -> str:
         result.assert_exit_code()
         return result.stdout
 
-    def _hugepages_init(self, node: Node) -> None:
-        mount = node.tools[Mount]
-        mount.mount(disk_name="nodev", point="/mnt/huge", type="hugetlbfs")
-        mount.mount(
-            disk_name="nodev",
-            point="/mnt/huge-1G",
-            type="hugetlbfs",
-            options="pagesize=1G",
-        )
+    @TestCaseMetadata(
+        description="""
+           Tests a more realistic sender/forwarder/receiver setup.
+        """,
+        priority=2,
+        requirement=simple_requirement(
+            min_core_count=8,
+            min_nic_count=2,
+            network_interface=Sriov,
+            min_count=3,
+        ),
+    )
+    def check_3node_forwarding(self, environment: Environment, log: Logger) -> None:
+        # TODO: stub test, just submitting refactoring PR
+        pass
+
+
+def _hugepages_init(node: Node) -> None:
+    mount = node.tools[Mount]
+    mount.mount(disk_name="nodev", point="/mnt/huge", type="hugetlbfs")
+    mount.mount(
+        disk_name="nodev",
+        point="/mnt/huge-1G",
+        type="hugetlbfs",
+        options="pagesize=1G",
+    )
 
-    def _hugepages_enable(self, node: Node, log: Logger) -> None:
-        echo = node.tools[Echo]
-        echo.write_to_file(
-            "1024",
-            "/sys/devices/system/node/node0/hugepages/hugepages-2048kB/nr_hugepages",
-            sudo=True,
-        )
-        echo.write_to_file(
-            "1",
-            "/sys/devices/system/node/node0/hugepages/hugepages-1048576kB/nr_hugepages",
-            sudo=True,
+
+def _hugepages_enable(node: Node) -> None:
+    echo = node.tools[Echo]
+    echo.write_to_file(
+        "1024",
+        "/sys/devices/system/node/node0/hugepages/hugepages-2048kB/nr_hugepages",
+        sudo=True,
+    )
+    echo.write_to_file(
+        "1",
+        "/sys/devices/system/node/node0/hugepages/hugepages-1048576kB/nr_hugepages",
+        sudo=True,
+    )
+
+
+class DpdkTestKit:
+    def __init__(self, node: Node, log: Logger) -> None:
+        self._node = node
+        self._log = log
+
+    def initialize(self) -> None:
+        network_interface_feature = self._node.features[NetworkInterface]
+        sriov_is_enabled = network_interface_feature.is_enabled_sriov()
+        self._log.info(
+            f"Node[{self._node.name}] Verify SRIOV is enabled: {sriov_is_enabled}"
         )
+        assert_that(sriov_is_enabled).described_as(
+            f"SRIOV was not enabled for this test node ({self._node.name})"
+        ).is_true()
+
+        # dump some info about the pci devices before we start
+        lspci = self._node.tools[Lspci]
+        self._log.info(f"Node[{self._node.name}] LSPCI Info:\n{lspci.run().stdout}\n")
+        # enable hugepages (required by dpdk)
+        _hugepages_init(self._node)
```

Comment:
The init and enable called together always, can they be merged?

Reply:
init_hugepages and enable_hugepages may be good names.

Reply:
They are at the moment but won't always be, I'd prefer to keep them separate for now